### PR TITLE
Add details about updating item metadata in MSBuild

### PR DESCRIPTION
**Pull request by cdmihai**
Should be merged only after the following PR from MSBuild makes it into Visual Studio: https://github.com/microsoft/msbuild/pull/4975

### CODE REVIEW

**Comment by PRMerger16** [comment]
@cdmihai : Thanks for your contribution! The author(s) have been notified to review your proposed change. @kempb

**Comment by PRMerger17** [comment]
@cdmihai : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by cdmihai** [comment]
@rainersigwald: Added documentation for updating metadata, including the new qualified metadata references.

File: docs/msbuild/msbuild-items.md [hunk 1]
```diff
@@ -330,6 +330,259 @@ Output:
 -->
 ```
 
+##  <a name="BKMK_UpdatingMetadataOnExistingItemsOutsideTargets"></a> Updating metadata on items in an ItemGroup outside of a Target
```
Comment (ghogen) [inline docs/msbuild/msbuild-items.md]: We generate anchor links automatically now - anchor will be #updating-metadata-on-items-in-an-itemgroup-outside-of-a-target .
```suggestion
##  Updating metadata on items in an ItemGroup outside of a Target
```

File: docs/msbuild/msbuild-items.md [hunk 2]
```diff
@@ -330,6 +330,259 @@ Output:
...
+- Unqualified metadata (%(M)) binds to the item type being updated (e.g. `Item1` in above example). Qualified metadata (e.g. `%(Item2.Color)`) binds inside the set of captured matching item types from the Update expression.
+- If an item matches multiple times within and between multiple referenced items:
+  - The last occurrence from each referenced item type gets captured (so one captured item per item type).
+  - This matches the behaviour of task item batching under targets.
+- Where one can put %() references:
+  - Metadata
+  - Metadata conditions
+- Metadata name matching is case insensitive.
+
+##  <a name="BKMK_UpdatingMetadataOnExistingItemsInsideTargets"></a> Updating metadata on items in an ItemGroup of a Target
```
Comment (ghogen) [inline docs/msbuild/msbuild-items.md]:
```suggestion
##  Updating metadata on items in an ItemGroup of a Target
```

File: docs/msbuild/msbuild-items.md [hunk 3]
```diff
@@ -330,6 +330,259 @@ Output:
...
+    Size: large
+    Color:
+    Material: paper
+    Price: 20
+    Model: 2020
+-->
+```
+
+Remarks:
+- Unqualified metadata (%(M)) binds to the item type being updated (e.g. `Item1` in above example). Qualified metadata (e.g. `%(Item2.Color)`) binds inside the set of captured matching item types from the Update expression.
```
Comment (ghogen) [inline docs/msbuild/msbuild-items.md]: We're not supposed to use `e.g.` or `i.e.` in docs.
```suggestion
- Unqualified metadata (%(M)) binds to the item type being updated (`Item1` in above example). Qualified metadata (`%(Item2.Color)`) binds inside the set of captured matching item types from the Update expression.
```

**Comment by PRMerger17** [comment]
@cdmihai : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger17** [comment]
@cdmihai : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by rainersigwald** [comment]
@ghogen, this is the new 16.6 feature that we talked about the other day, so we should probably retarget this to the internal 16.6 branch. It's tied to https://github.com/microsoft/msbuild/pull/4975.

**Comment by ghogen** [comment]
@rainersigwald @cdmihai OK, let's close this here then and resubmit the changes internally. I created a release-16.6 branch just now for us to use.

**Pull request closed without merging**